Mathlib/FieldTheory/JacobsonNoether.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2024 F. Nuccio, H. Zheng, W. He, S. Wu, Y. Yuan, W. Jiao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Filippo A. E. Nuccio, Huanyu Zheng, Sihan Wu, Wanyi He, Weichen Jiao, Yi Yuan
+-/
+import Mathlib.Algebra.CharP.LinearMaps
+import Mathlib.Algebra.Field.Power
+import Mathlib.FieldTheory.PurelyInseparable
+
+/-!
+# The Jacobson-Noether theorem
+
+This file contains a proof of the Jacobson-Noether theorem and some auxiliary lemmas.
+Here we discuss different cases of characteristics of
+the noncommutative division algebra `D` with center `k`.
+
+## Main Results
+
+- `Jacobson_Noether` : Let `D` be a noncommutative division algebra with center `k`.
+  If `D` is algebraic over `k`,
+  then there exists an element of `D \ k` which is separable over `k`.
+
+## Notations
+
+- `D` is a noncommutative division algebra
+- `k` is the center of `D`
+
+## Implementation Notes
+
+## Reference
+* <https://ysharifi.wordpress.com/2011/09/30/the-jacobson-noether-theorem/>
+-/
+
+namespace JacobsonNoether
+
+variable {D : Type*} [DivisionRing D]
+
+open LinearMap
+
+local notation3 "k" => (Subring.center D)
+private def δ : D → D →ₗ[k] D := mulLeft k - mulRight k
+
+private lemma δ_def (a x : D) : δ a x = mulLeft k a x - mulRight k a x := rfl
+
+private lemma δ_def' (a : D) : δ a = mulLeft k a - mulRight k a := rfl
+
+private lemma l_pow (a : D) (n : ℕ) : ∀ x : D, ((mulLeft k a) ^ n).1 x = (a ^ n) * x := by
+  intro x
+  rw [coe_toAddHom, pow_apply]
+  induction n with
+  | zero => rw [Function.iterate_zero, id_eq, pow_zero, one_mul]
+  | succ _ h => rw [Function.iterate_succ', Function.comp_apply, h,
+    pow_succ', mul_assoc, mulLeft_apply]
+
+private lemma r_pow (a : D) (n : ℕ) : ∀ x : D, ((mulRight k a) ^ n).1 x = x * (a ^ n) := by
```

Comment:
```suggestion
private lemma r_pow (a : D) (n : ℕ) : ∀ x : D, ((mulRight k a) ^ n) x = x * (a ^ n) := by
```
Type class inference is able to see through the projection from this ring hom down to a `AddHom`, so please remove this `.1`; you can then remove the `coe_toAddHom` from the `rw`, and do the same for `r_pow`. In the applications around line 112, it suffices again to remove the projection `.1`.